### PR TITLE
refactor: use helper from controller runtime

### PR DESCRIPTION
**Pull request by flavio**
Use a more concise way to handle "not found" errors that is made possible by using a controller runtime helper.

### CODE REVIEW

**Review by viccuad — APPROVED** [review]

**Comment by flavio** [comment]
@jvanz, @viccuad : I've updated golanglint-ci and then fixed all the warnings via 8d936333779b11f9cb2f1314341325b0f3af80a3. Please take a look at the commit description to see the reasons of the changes I've made

**Comment by flavio** [comment]
**Note:** this cannot be merged until the 1.12 release is done

**Comment by jvanz** [comment]
It seems there is something wrong, because tests and linter are failing. =(

**Comment by flavio** [comment]
@jvanz I'm a bit stuck with other tasks, would you be so kind to take this over once your PR about the owner reference is merged? :pray:

**Comment by jvanz** [comment]
@kubewarden/kubewarden-developers , I've updated this PR rebasing it on top of main and did some fixes to make the CI happy. To make all our CI green I had to update the controller-gen binary used. Otherwise, we get a nil pointer deref. That's why you can see so many line changes, I've commited the result of the generator after updating it. Let me know if you're fine with that. Otherwise, I can update the generator in another PR or commit the updated manifest files in another PR.

**Review by jvanz — APPROVED** [review]

**Review by flavio — COMMENTED** [review]
I'm fine with the update of `controller-gen` binary. Everything looks good to me, I left some minor comments.

**Review by fabriziosestito — APPROVED** [review]
LGTM apart from the changes requested

File: controllers/clusteradmissionpolicy_controller_test.go [hunk 1]
```diff
@@ -74,7 +72,7 @@ var _ = Describe("ClusterAdmissionPolicy controller", func() {
 				Expect(validatingWebhookConfiguration.Annotations[constants.WebhookConfigurationPolicyNameAnnotationKey]).To(Equal(policyName))
 				Expect(validatingWebhookConfiguration.Annotations[constants.WebhookConfigurationPolicyNamespaceAnnotationKey]).To(BeEmpty())
 				Expect(validatingWebhookConfiguration.Webhooks).To(HaveLen(1))
-				Expect(validatingWebhookConfiguration.Webhooks[0].ClientConfig.Service.Name).To(Equal(fmt.Sprintf("policy-server-%s", policyServerName)))
+				Expect(validatingWebhookConfiguration.Webhooks[0].ClientConfig.Service.Name).To(Equal("policy-server-" + policyServerName))
```
[the remark this answers is not in the document]
Comment (flavio) [reply controllers/clusteradmissionpolicy_controller_test.go]: done

File: controllers/clusteradmissionpolicy_controller_test.go [hunk 2]
```diff
@@ -195,7 +193,7 @@ var _ = Describe("ClusterAdmissionPolicy controller", func() {
 
 			By("reconciling the MutatingWebhookConfiguration to its original state")
 			Eventually(func() (*admissionregistrationv1.MutatingWebhookConfiguration, error) {
-				return getTestMutatingWebhookConfiguration(fmt.Sprintf("clusterwide-%s", policyName))
+				return getTestMutatingWebhookConfiguration("clusterwide-%s" + policyName)
```
[the remark this answers is not in the document]
Comment (flavio) [reply controllers/clusteradmissionpolicy_controller_test.go]: done

File: internal/pkg/admission/policy-server-ca-secret_test.go [hunk 3]
```diff
@@ -88,7 +86,7 @@ func TestFetchOrInitializePolicyServerSecret(t *testing.T) {
 	caSecret := &corev1.Secret{Data: map[string][]byte{constants.PolicyServerCARootCACert: admissionregCA.CaCert, constants.PolicyServerCARootPrivateKeyCertName: x509.MarshalPKCS1PrivateKey(admissionregCA.CaPrivateKey)}}
 
 	//nolint:unparam
-	generateCertFunc := func(ca []byte, commonName string, extraSANs []string, CAPrivateKey *rsa.PrivateKey) ([]byte, []byte, error) {
+	generateCertFunc := func(_ []byte, _ string, _ []string, _ *rsa.PrivateKey) ([]byte, []byte, error) {
```
[the remark this answers is not in the document]
Comment (flavio) [reply internal/pkg/admission/policy-server-ca-secret_test.go:89]: fixed

**Comment by jvanz** [comment]
@kubewarden/kubewarden-developers I've updated this branch rebasing it on top of main and squashing some commits to reduce the number of them.

**Pull request merged**